### ISSUE TITLE
乐视网视频无法合并

### ISSUE THREAD

**Issue by huangchen1031** [soimort/you-get#1047]
请使用`--debug`运行，并将输出粘贴在下面:

```
Merging video parts...
.\格林童话剧场 01[00].mp4
you-get: version 0.4.365, a tiny downloader that scrapes the web.
you-get: ['http://www.le.com/ptv/vplay/1650028.html']
Traceback (most recent call last):
  File "c:\users\晨\appdata\local\programs\python\python35\lib\runpy.py", line 170, in _run_module_as_main
    "__main__", mod_spec)
  File "c:\users\晨\appdata\local\programs\python\python35\lib\runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "C:\Users\晨\AppData\Local\Programs\Python\Python35\Scripts\you-get.exe\__main__.py", line 9, in <module>
  File "c:\users\晨\appdata\local\programs\python\python35\lib\site-packages\you_get\__main__.py", line 92, in main
    main(**kwargs)
  File "c:\users\晨\appdata\local\programs\python\python35\lib\site-packages\you_get\common.py", line 1274, in main
    script_main('you-get', any_download, any_download_playlist, **kwargs)
  File "c:\users\晨\appdata\local\programs\python\python35\lib\site-packages\you_get\common.py", line 1187, in script_main
    download_main(download, download_playlist, args, playlist, output_dir=output_dir, merge=merge, info_only=info_only, json_output=json_output, caption=caption)
  File "c:\users\晨\appdata\local\programs\python\python35\lib\site-packages\you_get\common.py", line 1030, in download_main
    download(url, **kwargs)
  File "c:\users\晨\appdata\local\programs\python\python35\lib\site-packages\you_get\common.py", line 1267, in any_download
    m.download(url, **kwargs)
  File "c:\users\晨\appdata\local\programs\python\python35\lib\site-packages\you_get\extractors\le.py", line 137, in letv_download
    letv_download_by_vid(vid, title=title, output_dir=output_dir, merge=merge, info_only=info_only,**kwargs)
  File "c:\users\晨\appdata\local\programs\python\python35\lib\site-packages\you_get\extractors\le.py", line 98, in letv_download_by_vid
    download_urls(urls, title, ext, size, output_dir=output_dir, merge=merge)
  File "c:\users\晨\appdata\local\programs\python\python35\lib\site-packages\you_get\common.py", line 765, in download_urls
    concat_mp4(parts, output_filepath)
  File "c:\users\晨\appdata\local\programs\python\python35\lib\site-packages\you_get\processor\join_mp4.py", line 914, in concat_mp4
    merge_mp4s(mp4s, output)
  File "c:\users\晨\appdata\local\programs\python\python35\lib\site-packages\you_get\processor\join_mp4.py", line 876, in merge_mp4s
    mp4s = list(map(read_mp4, ins))
  File "c:\users\晨\appdata\local\programs\python\python35\lib\site-packages\you_get\processor\join_mp4.py", line 717, in read_mp4
    atoms = parse_atoms(stream)
  File "c:\users\晨\appdata\local\programs\python\python35\lib\site-packages\you_get\processor\join_mp4.py", line 708, in parse_atoms
    atom = read_atom(stream)
  File "c:\users\晨\appdata\local\programs\python\python35\lib\site-packages\you_get\processor\join_mp4.py", line 700, in read_atom
    raise NotImplementedError('%s: %d' % (type, left))
NotImplementedError: b'\x00\x00\xb0\r': 1195376648
```

如果您有其他附言，例如问题只在某个视频发生，或者是一般性讨论或者提出新功能，请在下面添加;或者您可以卖个萌:

> 大神~请允许我卖个萌~~~
> [舔 :icecream:!]

**Comment by huangchen1031** [comment]
`C:\Users\晨>you-get -di http://www.le.com/ptv/vplay/1650028.html
[DEBUG] get_content: http://www.le.com/ptv/vplay/1650028.html
[DEBUG] get_content: http://api.letv.com/mms/out/video/playJson?id=1650028&platid=1&splatid=101&format=1&tkey=1205873813&domain=www.letv.com
Current Video Supports:
        --format 1300 <URL>
        --format 350 <URL>
        --format 1000 <URL>
[DEBUG] get_content: http://play.g3proxy.lecloud.com/vod/v2/MTEyLzUwLzEzL2xldHYtdXRzLzE0L3Zlcl8wMF8yMi0zMTM4OTQ4MTUtYXZjLTg4MDE0OC1hYWMtNjQwMDEtMTM4OTEyMC0xNjU1MTQwOTQtOGVjOWQwZDY1NzM2MjhkN2QwYzIwNTQ4NWQ4ZjUyYzEtMTQyNzUwMTA4MDA1OS5tcDQ=?b=953&mmsid=1835683&tm=1460194488&key=75f715264bdb10db463a10f6cdc2d95e&platid=1&splatid=101&playid=0&tss=ios&vtype=22&cvid=1119226653810&payff=0&pip=e08df80aa90ceab415e8195921eda725&ctv=pc&m3v=1&termid=1&format=1&hwtype=un&ostype=Linux&tag=letv&sign=letv&expect=3&tn=0.9993255265558734&pay=0&iscpn=f9051&rateid=1300
[DEBUG] get_content: http://119.188.174.134/112/50/13/letv-uts/14/ver_00_22-313894815-avc-880148-aac-64001-1389120-165514094-8ec9d0d6573628d7d0c205485d8f52c1-1427501080059.m3u8?crypt=86aa7f2e978&b=953&nlh=4096&nlt=60&bf=82&p2p=1&video_type=mp4&termid=1&tss=ios&platid=1&splatid=101&its=0&qos=4&fcheck=0&mltag=1&proxy=2008854160,2008855645,2007471075&uid=1928596617.rp&keyitem=GOw_33YJAAbXYE-cnQwpfLlv_b2zAkYctFVqe5bsXQpaGNn3T1-vhw..&ntm=1460205600&nkey=7a5a3eb65f8f4b417c51b33f1853b4c7&nkey2=d45839c44b311e99a4155ed29387e4e6&geo=CN-1-17-2&mmsid=1835683&tm=1460194488&key=75f715264bdb10db463a10f6cdc2d95e&playid=0&vtype=22&cvid=1119226653810&payff=0&ctv=pc&m3v=1&hwtype=un&ostype=Linux&tag=letv&sign=letv&tn=0.9993255265558734&pay=0&iscpn=f9051&rateid=1300&errc=0&gn=1170&buss=100&cips=114.244.12.137
Site:       Le.com
Title:      格林童话剧场 01
Type:       MPEG-4 video (video/mp4)
Size:       164.8 MiB (172800952 Bytes)`

**Comment by yxjxx** [comment]
```
(env3) ~/env3  you-get http://v.youku.com/v_show/id_XODgxODkzODI0.html
site:                优酷 (Youku)
title:               iOS线下分享《RunLoop》by 孙源@sunnyxx
stream:
    - format:        flvhd
      container:     flv
      video-profile: 标清
      size:          114.9 MiB (120450199 bytes)
    # download-with: you-get --format=flvhd [URL]

Downloading iOS线下分享《RunLoop》by 孙源@sunnyxx.flv ...
 100% (114.9/114.9MB) ├███████████████████████████████████████┤[15/15]  276 kB/s
Merging video parts... you-get: [error] oops, something went wrong.
you-get: don't panic, c'est la vie. please try the following steps:
you-get:   (1) Rule out any network problem.
you-get:   (2) Make sure you-get is up-to-date.
you-get:   (3) Check if the issue is already known, on
you-get:         https://github.com/soimort/you-get/wiki/Known-Bugs
you-get:         https://github.com/soimort/you-get/issues
you-get:   (4) Run the command with '--debug' option,
you-get:       and report this issue with the full output.
```

**Comment by crystoneme** [comment]
same:

you-get --debug http://www.le.com/ptv/vplay/23035173.html
[DEBUG] get_content: http://www.le.com/ptv/vplay/23035173.html
[DEBUG] get_content: http://api.letv.com/mms/out/video/playJson?id=23035173&platid=1&splatid=101&format=1&tkey=550971541&domain=www.letv.com
Current Video Supports:
	--format 1000 <URL>
	--format 1300 <URL>
	--format 720p <URL>
	--format 350 <URL>
	--format 1080p <URL>
[DEBUG] get_content: http://play.g3proxy.lecloud.com/vod/v2/MTQ3LzQwLzYwL2xldHYtdXRzLzE0L3Zlcl8wMF8yMi0zMjE4NzAwMDctYXZjLTMwMDc3OTAtYWFjLTEyODAwOC0zMjM2NDAtMTI3MzQ0MjUzLThiMzI5OGVkYzRlODk0MGI5ZDBkZTBlODNhZGFkMDAzLTE0MzU2NTAxMTUxNTEubXA0?b=3147&mmsid=32594656&tm=1460732274&key=de61da2e285c09570e586aadff29092c&platid=1&splatid=101&playid=0&tss=ios&vtype=52&cvid=834270726097&payff=0&pip=1b75fe24af7d41dd61456b86b7b03cd1&ctv=pc&m3v=1&termid=1&format=1&hwtype=un&ostype=Linux&tag=letv&sign=letv&expect=3&tn=0.681604155398427&pay=0&iscpn=f9051&rateid=1080p
[DEBUG] get_content: http://183.131.81.25/147/40/60/letv-uts/14/ver_00_22-321870007-avc-3007790-aac-128008-323640-127344253-8b3298edc4e8940b9d0de0e83adad003-1435650115151.m3u8?crypt=84aa7f2e3000&b=3147&nlh=4096&nlt=60&bf=76&p2p=1&video_type=mp4&termid=1&tss=ios&platid=1&splatid=101&its=0&qos=3&fcheck=0&mltag=100&proxy=3078830680,3078849080,1032384110&uid=1943927398.rp&keyitem=GOw_33YJAAbXYE-cnQwpfLlv_b2zAkYctFVqe5bsXQpaGNn3T1-vhw..&ntm=1460743200&nkey=73ce7be1af3e83e7865d74d0d27c43af&nkey2=46001a0b10c7d70070ce09796ebdd55d&geo=CN-11-140-1&payff=0&sign=letv&playid=0&hwtype=un&uip=115.221.250.102&tag=letv&vtype=52&tm=1460732274&tn=0.681604155398427&ctv=pc&ostype=Linux&cvid=834270726097&g3proxy_ercode=0&pay=0&iscpn=f9051&mmsid=32594656&rateid=1080p&g3proxy_tm=1460732250&key=de61da2e285c09570e586aadff29092c&errc=0&gn=1062&buss=100&cips=115.221.250.102&m3v=1
Site:       Le.com
Title:      山阴路的夏天（直播实录：李志2015巡演北京站）
Type:       MPEG-4 video (video/mp4)
Size:       124.8 MiB (130866988 Bytes)

Downloading 山阴路的夏天（直播实录：李志2015巡演北京站）.mp4 ...
 100% (124.8/124.8MB) ├█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████┤[21/21]   15 MB/s
Merging video parts... you-get: version 0.4.365, a tiny downloader that scrapes the web.
you-get: ['http://www.le.com/ptv/vplay/23035173.html']
Traceback (most recent call last):
  File "/usr/local/bin/you-get", line 11, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.5/site-packages/you_get/__main__.py", line 92, in main
    main(**kwargs)
  File "/usr/local/lib/python3.5/site-packages/you_get/common.py", line 1274, in main
    script_main('you-get', any_download, any_download_playlist, **kwargs)
  File "/usr/local/lib/python3.5/site-packages/you_get/common.py", line 1187, in script_main
    download_main(download, download_playlist, args, playlist, output_dir=output_dir, merge=merge, info_only=info_only, json_output=json_output, caption=caption)
  File "/usr/local/lib/python3.5/site-packages/you_get/common.py", line 1030, in download_main
    download(url, **kwargs)
  File "/usr/local/lib/python3.5/site-packages/you_get/common.py", line 1267, in any_download
    m.download(url, **kwargs)
  File "/usr/local/lib/python3.5/site-packages/you_get/extractors/le.py", line 137, in letv_download
    letv_download_by_vid(vid, title=title, output_dir=output_dir, merge=merge, info_only=info_only,**kwargs)
  File "/usr/local/lib/python3.5/site-packages/you_get/extractors/le.py", line 98, in letv_download_by_vid
    download_urls(urls, title, ext, size, output_dir=output_dir, merge=merge)
  File "/usr/local/lib/python3.5/site-packages/you_get/common.py", line 762, in download_urls
    ffmpeg_concat_mp4_to_mp4(parts, output_filepath)
  File "/usr/local/lib/python3.5/site-packages/you_get/processor/ffmpeg.py", line 174, in ffmpeg_concat_mp4_to_mp4
    subprocess.check_call(params)
  File "/usr/local/Cellar/python3/3.5.1/Frameworks/Python.framework/Versions/3.5/lib/python3.5/subprocess.py", line 584, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['ffmpeg', '-loglevel', 'quiet', '-f', 'concat', '-y', '-i', './山阴路的夏天（直播实录：李志2015巡演北京站）.mp4.txt', '-c', 'copy', './山阴路的夏天（直播实录：李志2015巡演北京站）.mp4']' returned non-zero exit status 1

**Comment by haodong** [comment]
exactly same error as above.

**Comment by haodong** [comment]
temporarily solved by adding `params = [FFMPEG] + LOGLEVEL + ['-f', 'concat', "-safe", "0", '-y', '-i']` in File "/usr/local/lib/python3.5/site-packages/you_get/processor/ffmpeg.py", line 129.

**Comment by yaccai** [comment]
不能合并好像是因为文件路径名有问题，手动ffmpeg合并，也有错误提示 unsafe file name。ffmpeg合并时的解决办法：去掉文件名中的不安全字符如[]等，或者加上-safe 0选项。you-get应该也是类似的原因吧

**Comment by huangchen1031** [comment]
我还以为是文件个数太多了呢~ 我在乐视下载其他视频的时候有的不会出现问题

**Comment by soimort** [comment]
Resolved as #1023.
